src/Symfony/Component/DependencyInjection/Compiler/ResolveDefinitionInheritancePass.php

```diff
@@ -27,7 +27,7 @@ protected function processValue($value, $isRoot = false)
             return parent::processValue($value, $isRoot);
         }
 
-        $class = $value instanceof ChildDefinition ? $this->resolveDefinition($value) : $value->getClass();
+        $class = $value->getClass();
```

Comment:
This entire class/file could/should be renamed (`ResolveInstanceofPass`) - I didn't do it to keep the diff more manageable